File: packages/object-detection/android/src/main/java/io/capawesome/capacitorjs/plugins/mlkit/objectdetection/classes/options/ProcessImageOptions.java
package io.capawesome.capacitorjs.plugins.mlkit.objectdetection.classes.options;

import androidx.annotation.NonNull;
import com.getcapacitor.PluginCall;
import io.capawesome.capacitorjs.plugins.mlkit.objectdetection.ObjectDetectionPlugin;

public class ProcessImageOptions {

    @NonNull
    private final String path;

    private final boolean shouldEnableClassification;

    private final boolean shouldEnableMultipleObjects;

    public ProcessImageOptions(@NonNull PluginCall call) throws Exception {
        String path = call.getString("path", null);
        if (path == null) {
            throw new Exception(ObjectDetectionPlugin.ERROR_PATH_MISSING);
        }
        this.path = path;
        this.shouldEnableClassification = call.getBoolean("shouldEnableClassification", false);
        this.shouldEnableMultipleObjects = call.getBoolean("shouldEnableMultipleObjects", false);
    }

    @NonNull
    public String getPath() {
        return path;
    }

    public boolean isClassificationEnabled() {
        return shouldEnableClassification;
    }

    public boolean isMultipleObjectsEnabled() {
        return shouldEnableMultipleObjects;
    }
}

File: packages/object-detection/ios/Plugin/Classes/Options/ProcessImageOptions.swift
import Foundation
import Capacitor

@objc public class ProcessImageOptions: NSObject {
    private var path: String
    private var shouldEnableClassification: Bool
    private var shouldEnableMultipleObjects: Bool

    init(_ call: CAPPluginCall) throws {
        guard let path = call.getString("path") else {
            throw CustomError.pathMissing
        }
        self.path = path
        self.shouldEnableClassification = call.getBool("shouldEnableClassification", false)
        self.shouldEnableMultipleObjects = call.getBool("shouldEnableMultipleObjects", false)
    }

    func getPath() -> String {
        return path
    }

    func isClassificationEnabled() -> Bool {
        return shouldEnableClassification
    }

    func isMultipleObjectsEnabled() -> Bool {
        return shouldEnableMultipleObjects
    }
}
